umh-core/pkg/diagnosis/judge.go

```
@@ -0,0 +1,241 @@
+
+	return clamp01((worse(f.Value, m) - fire) / (worse(m.Worst, m) - fire))
+}
+
+// Rank orders the signals that fired so the caller can show the main reason
+// first: tier ascending (a lower tier outranks a higher one), then severity
+// descending, then external attribution first, then the signal's table index.
+// It sorts in place and returns the same slice.
+func Rank(fired []Fired) []Fired {
+	sort.Slice(fired, func(i, j int) bool {
+		a, b := fired[i], fired[j]
+		if a.Identity.Tier != b.Identity.Tier {
```

golangci-lint (2.12.2)

[error] QF1008: could remove embedded field "Identity" from selector

(staticcheck)